.github/workflows/snapshot.yml

```diff
@@ -0,0 +1,34 @@
+name: "snapshot"
+
+on:
+  pull_request:
+    paths-ignore:
+      - "!**.md"
+  push:
+    branches:
+      - "main"
+
+env:
+  DOCKER_BUILDKIT: 1
+  FORCE_COLOR: 1
+
+jobs:
+  version:
+    name: "Generate snapshot id"
+    runs-on: ubuntu-latest
+    outputs:
+      id: ${{ steps.version.outputs.id }}
+    steps:
+      - name: Generate snapshot version
+        id: version
+        uses: fluencelabs/github-actions/generate-snapshot-id@main
+
+  snapshot:
+    needs: version
+    permissions:
+      contents: read
+      id-token: write
+    uses: ./.github/workflows/container.yml
+    with:
+      image: "docker.fluence.dev/distro"
```

Comment:
hmmm, maybe `docker.fluence.dev/rust-peer:distro-$BRANCHNAME`? it's kinda strange to have basically the same image in `distro` and `rust-peer` 🤔 

I don't think "source" (from which repo it comes) should matter, only artifact semantics matters.

Reply:
Well knowing source is good. We only have rust-peer images in our distro :) Maybe something like `docker.fluence.dev/rust-peer-distro/rust-peer:branchetc` is better?

Reply:
I like to see repo name in image name. It will be easier to clean up later

Reply:
no `/distro` is better. well, ok, if you insist :) Let's keep status quo and see how it goes.